### ISSUE TITLE
Suggestion: an operator similar to magrittr's %<>% 

### ISSUE THREAD

**Issue by bleearmstrong** [kieferk/dfply#9]
A common idiom in dplyr is something like

```
df = df %>% (stuff)...
```

which can be simplified with

```
df %<>% (stuff)
```

I've looked around to see there's a way to overload `<>`, and I'm not sure there is (I think this gets intrepreted as `__neq__`, but then that also applied to `!=`).

At any rate, I think this would be very useful (essentially just doing things inplace instead of with copies)

It might be worth noting that Hadley has purposely left this out of dplyr, because (I believe) he's sort of opposed to doing things inplace.

**Comment by kieferk** [comment]
Interesting, I'm not sure the best way to do it but I will look into overloading an operator to do this. I'll keep you posted.

**Comment by kieferk** [comment]
So apparently the `>>=` operator in python will already do this for you. So if we have a dataframe `a`:

```python
    a  b
0  1  3
1  2  6
2  3  9
```

Then typing `a >>= mutate(c=X.a+X.b)` is equivalent to typing `a = a >> mutate(c=X.a+X.b)`. You can see that the command returns nothing if you enter it.

```python
a >>= mutate(c=X.a+X.b)
```

But then if you print out `a` you can see that the new column is there.

```python
    a  b   c
0  1  3   4
1  2  6   8
2  3  9  12
```

Unfortunately, because of precedence and order of operation stuff, this only works with one function following the `>>=` operator. I have pushed a branch called "feature/inplace-piping" that changes the code in the `pipe` decorator to make this work. Essentially it adds in an `__rshift__` implementation to pipe and instead of waiting for the dataframe to be passed through on the left, it chains together the pipe functions into a list until it reaches the dataframe, at which point `__rrshift__` is called and the piping functions are evaluated in order.

I really haven't done any testing on this to make sure it works aside from really simple messing around in the console and making sure the unit tests pass. I'm also not sure if this is a particularly elegant solution and welcome any comments or suggestions.

Cheers

**Comment by kieferk** [comment]
I implemented these changes in v0.2.3 which is live now on the master branch. Let me know if you run into any problems.
